src/types/context.ts

```diff
@@ -6,14 +6,14 @@ import { createAdapters } from "../adapters";
 import { Env } from "./env";
 import { Logs } from "@ubiquity-dao/ubiquibot-logger";
 
-export type SupportedEventsU = "issue_comment.created" | "issues.assigned" | "pull_request.opened";
+export type SupportedEventsU = "issue_comment.created" | "issues.assigned" | "pull_request.opened" | "issues.unassigned";
 
 export type SupportedEvents = {
   [K in SupportedEventsU]: K extends WebhookEventName ? WebhookEvent<K> : never;
 };
 
 export function isContextCommentCreated(context: Context): context is Context<"issue_comment.created"> {
```

Comment:
```suggestion
export function isIssueEvent(context: Context): context is Context<"issues.opened"> {
```
I feel like `issue_comment.created` could lead to bugs because it would make types seem like there is a comment present even though you don't know if it is.
Even more type safe would be to have a type guard for every event type.

But if a function is shared for many events and you want to simply know if issue is present in the payload you can just use if statement and typescript will automatically infer it

Reply:
Changed in https://github.com/ubiquibot/command-start-stop/pull/46/commits/2f975345a7f1639cd62ea28d0e2e4dde1f31376d